nima/webserver/webserver/src/test/java/io/helidon/nima/webserver/builder/ContentEncodingContextTest.java

```diff
@@ -0,0 +1,95 @@
+import io.helidon.nima.webserver.WebServer;
+import io.helidon.nima.webserver.http.Handler;
+import io.helidon.nima.webserver.http.HttpRouting;
+import io.helidon.nima.webserver.http.ServerRequest;
+import io.helidon.nima.webserver.http.ServerResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.NoSuchElementException;
+import java.util.function.Consumer;
+
+public class ContentEncodingContextTest {
+    class CustomizedEncodingContext implements ContentEncodingContext {
+        int ACCEPT_ENCODING_COUNT = 0;
+        int NO_ACCEPT_ENCODING_COUNT = 0;
+
+        ContentEncodingContext contentEncodingContext = ContentEncodingContext.create();
+
+        @Override
+        public boolean contentEncodingEnabled() {
+            return contentEncodingContext.contentEncodingEnabled();
+        }
+
+        @Override
+        public boolean contentDecodingEnabled() {
+            return contentEncodingContext.contentDecodingEnabled();
+        }
+
+        @Override
+        public boolean contentEncodingSupported(String encodingId) {
+            return contentEncodingContext.contentEncodingSupported(encodingId);
+        }
+
+        @Override
+        public boolean contentDecodingSupported(String encodingId) {
+            return contentEncodingContext.contentDecodingSupported(encodingId);
+        }
+
+        @Override
+        public ContentEncoder encoder(String encodingId) throws NoSuchElementException {
+            return contentEncodingContext.encoder(encodingId);
+        }
+
+        @Override
+        public ContentDecoder decoder(String encodingId) throws NoSuchElementException {
+            return contentEncodingContext.decoder(encodingId);
+        }
+
+        @Override
+        public ContentEncoder encoder(Headers headers) {
+            if (headers.contains(Http.Header.ACCEPT_ENCODING)) {
+                ACCEPT_ENCODING_COUNT++;
+            } else {
+                NO_ACCEPT_ENCODING_COUNT++;
+            }
+            return contentEncodingContext.encoder(headers);
+        }
+    }
+
+    @Test
+    void testCustomizeContentEncodingContext() {
+        CustomizedEncodingContext encodingContext = new CustomizedEncodingContext();
+
+        WebServer ws = WebServer.builder()
+                .contentEncodingContext(encodingContext)
+                .routing(new Consumer<HttpRouting.Builder>() {
+                    @Override
+                    public void accept(HttpRouting.Builder rules) {
+                        rules.get("/hello", new Handler() {
+                            @Override
+                            public void handle(ServerRequest req, ServerResponse res) throws Exception {
+                                res.send("hello nima");
+                            }
+                        });
+                    }
+                })
+                .start();
+
+        try (Http1ClientResponse response = Http1Client.builder().build().get()
```

Comment:
When you use `@ServerTest`, you can get an instance of client already configured for the current server.